src/Microsoft.Data.Sqlite/Interop/NativeMethods`.cs

```diff
@@ -0,0 +1,200 @@
+        }
+
+        public static string sqlite3_column_text(Sqlite3StmtHandle stmt, int iCol) => MarshalEx.PtrToStringUTF8(sqlite3_column_text_raw(stmt, iCol));
+
+        public static string sqlite3_column_name(Sqlite3StmtHandle stmt, int iCol) => MarshalEx.PtrToStringUTF8(sqlite3_column_name_raw(stmt, iCol));
+
+        public static string sqlite3_column_decltype(Sqlite3StmtHandle stmt, int iCol) => MarshalEx.PtrToStringUTF8(sqlite3_column_decltype_raw(stmt, iCol));
+
+        public static string sqlite3_db_filename(Sqlite3Handle db, string zDbName)
+        {
+            var ptr = MarshalEx.StringToHGlobalUTF8(zDbName);
+            try
+            {
+                return MarshalEx.PtrToStringUTF8(sqlite3_db_filename(db, ptr));
+            }
+            finally
+            {
+                if (ptr != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(ptr);
+                }
+            }
+        }
+
+        public static string sqlite3_errstr(int rc) => MarshalEx.PtrToStringUTF8(sqlite3_errstr_raw(rc));
+
+        public static string sqlite3_errmsg(Sqlite3Handle db) => MarshalEx.PtrToStringUTF8(sqlite3_errmsg_raw(db));
+
+        public static string sqlite3_libversion() => MarshalEx.PtrToStringUTF8(sqlite3_libversion_raw());
+
+        public static int sqlite3_open_v2(string filename, out Sqlite3Handle ppDb, int flags, string vfs)
+        {
+            var zFilename = MarshalEx.StringToHGlobalUTF8(filename);
+            var zVfs = string.IsNullOrEmpty(vfs) ? IntPtr.Zero : MarshalEx.StringToHGlobalUTF8(vfs);
+            try
+            {
+                return sqlite3_open_v2(zFilename, out ppDb, flags, zVfs);
+            }
+            finally
+            {
+                if (zFilename != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(zFilename);
+                }
+                if (zVfs != IntPtr.Zero)
+                {
+                    Marshal.FreeHGlobal(zVfs);
+                }
+            }
+        }
+
+        public static int sqlite3_prepare_v2(Sqlite3Handle db, string zSql, out Sqlite3StmtHandle ppStmt, out string pzTail)
+        {
+            int nByte;
+            var zSqlPtr = MarshalEx.StringToHGlobalUTF8(zSql, out nByte);
+            try
+            {
+                // TODO: Something fancy with indexes?
+                IntPtr pzTailPtr;
+                var rc = sqlite3_prepare_v2(db, zSqlPtr, nByte, out ppStmt, out pzTailPtr);
+                pzTail = MarshalEx.PtrToStringUTF8(pzTailPtr);
+
+                return rc;
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(zSqlPtr);
+            }
+        }
+
+        public static int sqlite3_prepare_v2_blocking(Sqlite3Handle db, string zSql, out Sqlite3StmtHandle ppStmt, out string pzTail, int milliseconds)
+        {
+            sqlite3_busy_timeout(db, milliseconds);
+            return sqlite3_prepare_v2(db, zSql, out ppStmt, out pzTail);
```

Comment:
I'd prefer to keep NativeMethods 1:1 with the native APIs. I.e. Move this to SqliteCommand.